gpcontrib/arenadata_toolkit/arenadata_toolkit.c

```diff
@@ -1,4 +1,141 @@
+#include <dirent.h>
+#include <sys/stat.h>
+
 #include "postgres.h"
 #include "fmgr.h"
+#include "funcapi.h"
+#include "miscadmin.h"
+#include "storage/fd.h"
+#include "nodes/execnodes.h"
+#include "cdb/cdbvars.h"
+#include "utils/builtins.h"
+#include "utils/relfilenodemap.h"
+#include "utils/timestamp.h"
 
 PG_MODULE_MAGIC;
+
+#define should_skip(filename) \
+	(pg_strncasecmp(filename, "pg", 2) == 0 \
+	 || pg_strncasecmp(filename, "t_", 2) == 0 \
+	 || pg_strncasecmp(filename + strlen(filename) - 4, "_fsm", 4) == 0 \
+	 || pg_strncasecmp(filename + strlen(filename) - 3, "_vm", 3) == 0 \
+	 || pg_strncasecmp(filename + strlen(filename) - 5, "_init", 5) == 0)
+
+typedef struct
+{
+	char     *datpath;
+	DIR      *dirdesc;
+	TupleDesc tupdesc;
+} user_fctx_data;
+
+PG_FUNCTION_INFO_V1(adb_get_relfilenodes);
+Datum adb_get_relfilenodes(PG_FUNCTION_ARGS)
+{
+	Oid              datoid = MyDatabaseId;
+	Oid              tablespace_oid = MyDatabaseTableSpace;
```

Comment:
Where is the function argument?
```c
    Oid		tablespace_oid = PG_GETARG_OID(0);
```

Reply:
Using `PG_GETARG_OID(0)` instead of `MyDatabaseTableSpace` leads to `FalideAssetion`:
```
2023-06-05 10:56:59.343278 UTC,"gpadmin","postgres",p12090,th75643008,"172.17.0.2","54134",2023-06-05 10:56:59 UTC,0,con28,cmd4,seg0,slice4,,,sx1,"FATAL","XX000","Unexpected internal error (relpath.c:113)","FailedAssertion(""!(dbNode == 0)"", File: ""relpath.c"", Line: 113)","Process 12090 will wait for gp_debug_linger=120 seconds before termination.
Note that its locks and other resources will not be released until then.",,,,"select * from arenadata_toolkit.__db_files_current;",0,,"relpath.c",113,"Stack trace:
1    0xe31066 postgres errstart (elog.c:539)
2    0xe2f98d postgres ExceptionalCondition (assert.c:37)
3    0x1308d14 postgres GetDatabasePath (relpath.c:114)
4    0x7f4ee600088f arenadata_toolkit.so adb_get_relfilenodes + 0x90
```

Reply:
> Using `PG_GETARG_OID(0)` instead of `MyDatabaseTableSpace` leads to `FaliedAssetion`

This is no reason to get rid of the argument.